src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -508,9 +508,17 @@ int D4NFilterObject::D4NFilterReadOp::flush(const DoutPrefixProvider* dpp, rgw::
       std::pair<uint64_t, uint64_t> ofs_len_pair = it->second;
       uint64_t ofs = ofs_len_pair.first;
       uint64_t len = ofs_len_pair.second;
+      bool dirty = false;
+      /*
+      std::stringstream s;
+      utime_t ut(source->get_mtime());
+      ut.gmtime(s);
+      */
+      time_t creationTime = ceph::real_clock::to_time_t(source->get_mtime());
+
       std::string oid_in_cache = prefix + "_" + std::to_string(ofs) + "_" + std::to_string(len);
       ldpp_dout(dpp, 20) << "D4NFilterObject::" << __func__ << " calling update for offset: " << offset << " adjusted offset: " << ofs  << " length: " << len << " oid_in_cache: " << oid_in_cache << dendl;
-      source->driver->get_policy_driver()->get_cache_policy()->update(dpp, oid_in_cache, ofs, len, version, y);
+      source->driver->get_policy_driver()->get_cache_policy()->update(dpp, oid_in_cache, ofs, len, version, dirty, creationTime,  source->get_bucket()->get_owner(), y);
```

Comment:
in iterate() method, is reading of dirty blocks not handled/ allowed?

Reply:
In the next commit, I have updated the code to reflect this. I had missed these changes through the merge process.